api/service/discovery/service.go

```diff
@@ -104,7 +105,6 @@ func (s *Service) contactP2pPeers() {
 		case action := <-s.actionChan:
 			s.config.Actions[action.Name] = action.Action
 		case <-tick.C:
-			var err error
 			for g, a := range s.config.Actions {
 				if a == p2p.ActionPause {
 					// Recived Pause Message, to reduce the frequency of ping message to every 1 minute
```

Comment:
if you remove the logic below, this means there won't be any logic for <-tick.C; why do we keep it and why do we change to tick to a new ticker?

Reply:
There *is* logic though: The `sentPingMessage()` call.

Reply:
got it